frontend/packages/ceph-storage-plugin/src/utils/install.ts

```diff
@@ -113,22 +115,45 @@ export const countNodesPerZone = (nodes: NodeKind[]) =>
 export const nodesWithoutTaints = (nodes: NodeKind[]) =>
   nodes.filter((node: NodeKind) => hasOCSTaint(node) || hasNoTaints(node));
 
-export const isArbiterSC = (
-  sc: StorageClassResourceKind,
+const getSelectedNodes = (
+  scName: string,
   pvData: K8sResourceKind[],
   nodesData: NodeKind[],
-): boolean => {
-  const pvs: K8sResourceKind[] = getSCAvailablePVs(pvData, getName(sc));
+): NodeKind[] => {
+  const pvs: K8sResourceKind[] = getSCAvailablePVs(pvData, scName);
   const scNodeNames = getAssociatedNodes(pvs);
   const tableData: NodeKind[] = nodesData.filter(
     (node: NodeKind) =>
       scNodeNames.includes(getName(node)) ||
       scNodeNames.includes(node.metadata.labels?.['kubernetes.io/hostname']),
   );
+  return tableData;
+};
+
+export const isIntenalAttachedSC = (
+  scName: string,
+  pvData: K8sResourceKind[],
+  nodesData: NodeKind[],
+  failureDomain: string,
+): boolean => {
+  const tableData: NodeKind[] = getSelectedNodes(scName, pvData, nodesData);
+  if (failureDomain === 'rack') {
+    const uniqSelectedNodesRacks: Set<string> = new Set(tableData.map(getRack));
+    return _.compact([...uniqSelectedNodesRacks]).length >= MINIMUM_NODES;
+  }
+  return tableData.length >= MINIMUM_NODES;
+};
```

Comment:
We will check only for rack labels here and that they 
```suggestion
): boolean => {
  const tableData: NodeKind[] = getSelectedNodes(scName, pvData, nodesData);
    const uniqSelectedNodesRacks: Set<string> = new Set(tableData.map(getRack));
    return _.compact([...uniqSelectedNodesRacks]).length >= MINIMUM_NODES;
  }

};
```